src/jitlayers.cpp

```diff
@@ -287,6 +287,13 @@ extern "C" {
 
 namespace {
 
+// Use a local variable to hold the addresses to avoid generating a PLT
```

Comment:
It's `Procedure Linkage Table`. It's a function named `foo@plt` generated in the object file (`libjulia`) that will lazily lookup and jump to a function in another object file `libLLVM`. The GDB internal lookup function finds the PLT in our object file instead of the real one in `libLLVM` so it then lookup everything else in the wrong file and couldn't find it....